src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -383,54 +415,81 @@ bool GcInfoDecoder::IsSafePoint(UINT32 codeOffset)
     if(m_NumSafePoints == 0)
         return false;
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
```

Comment:
Removal of this ifdef is changing behavior for Linux x86. I guess that it is a bug fix. I do not see a reason why safepoint handling should be different for Linux x86.

cc @gbalykov

Reply:
I assumed we can't get here with x86 since we use a different gcinfo encoder.
If that is not the case and we can on Linux, I will put this back.

Overall, I think this `-1` hack is bogus and it would be better if we could stop doing/undoing this artificial adjustment, but it is not relevant to this PR. I just thought that `#else` is unreachable.